pkg/cleanup/cleanup.go

```diff
@@ -11,14 +16,18 @@ type Cleanup struct {
 	client    k8s.Client
 	log       logrus.FieldLogger
 	namespace string
+	prep      *prepare.Prepare
```

Comment:
Moving to a DNS package would allow us to not deal with backups, and only add/remove as needed.

I will use that as a basis to move this issue forward.